Lib/test/inspect_fodder.py

```diff
@@ -80,3 +80,14 @@ def contradiction(self):
     raise Exception()
 except:
     tb = sys.exc_info()[2]
+
+class Callable:
+    def __call__(self, *args):
```

Comment:
What if it is `async def __call__`?

Reply:
The `inspect` functions don't look at the `__call__` method, they support only instances of specific types (instances of `function`, `method` and `functools.partial`) and don't do duck typing. This was true before this PR and remains true after this PR.

Whether you agree with that or not, that's an independent issue and it shouldn't block this PR.

Reply:
A similar thing was already discussed at https://bugs.python.org/issue33261#msg315272 and following.

Reply:
> Whether you agree with that or not

As a maintainer of the inspect module I reserve the right to say otherwise. Please don't show this kind of attitude in communication with me or with any other core developer.

Reply:
As for "The inspect functions don't look at the __call__ method, they support only instances of specific types" I do indeed agree on that.

Reply:
> Please don't show this kind of attitude

I apologize for the misunderstanding. This was not meant as "attitude", just unfortunate wording on my part.

I meant to say "whether you agree or not with the fact that `inspect` functions don't look at `__call__`, discussing `__call__` should be a separate issue".